libs/common/src/services/folder/folder.service.ts

```diff
@@ -1,105 +1,78 @@
 import { CipherService } from "../../abstractions/cipher.service";
 import { CryptoService } from "../../abstractions/crypto.service";
 import { FolderService as FolderServiceAbstraction } from "../../abstractions/folder/folder.service.abstraction";
 import { I18nService } from "../../abstractions/i18n.service";
 import { StateService } from "../../abstractions/state.service";
-import { ServiceUtils } from "../../misc/serviceUtils";
 import { Utils } from "../../misc/utils";
 import { CipherData } from "../../models/data/cipherData";
 import { FolderData } from "../../models/data/folderData";
 import { Folder } from "../../models/domain/folder";
 import { SymmetricCryptoKey } from "../../models/domain/symmetricCryptoKey";
-import { TreeNode } from "../../models/domain/treeNode";
 import { FolderView } from "../../models/view/folderView";
 
-const NestingDelimiter = "/";
+const BroadcasterSubscriptionId = "FolderService";
 
 export class FolderService implements FolderServiceAbstraction {
+  private _folders: BehaviorSubject<Folder[]> = new BehaviorSubject([]);
+  private _folderViews: BehaviorSubject<FolderView[]> = new BehaviorSubject([]);
+
+  folders$ = this._folders.asObservable();
+  folderViews$ = this._folderViews.asObservable();
+
   constructor(
     private cryptoService: CryptoService,
     private i18nService: I18nService,
     private cipherService: CipherService,
-    private stateService: StateService
-  ) {}
-
-  async clearCache(userId?: string): Promise<void> {
-    await this.stateService.setDecryptedFolders(null, { userId: userId });
-  }
-
-  async encrypt(model: FolderView, key?: SymmetricCryptoKey): Promise<Folder> {
-    const folder = new Folder();
-    folder.id = model.id;
-    folder.name = await this.cryptoService.encrypt(model.name, key);
-    return folder;
-  }
-
-  async get(id: string): Promise<Folder> {
-    const folders = await this.stateService.getEncryptedFolders();
-    // eslint-disable-next-line
-    if (folders == null || !folders.hasOwnProperty(id)) {
-      return null;
-    }
-
-    return new Folder(folders[id]);
-  }
-
-  async getAll(): Promise<Folder[]> {
-    const folders = await this.stateService.getEncryptedFolders();
-    const response: Folder[] = [];
-    for (const id in folders) {
-      // eslint-disable-next-line
-      if (folders.hasOwnProperty(id)) {
-        response.push(new Folder(folders[id]));
+    private stateService: StateService,
+    private broadcasterService: BroadcasterService
+  ) {
+    this.stateService.activeAccount.subscribe(async (activeAccount) => {
+      if ((Utils.global as any).bitwardenContainerService == null) {
+        return;
       }
-    }
-    return response;
-  }
 
-  async getAllDecrypted(): Promise<FolderView[]> {
-    const decryptedFolders = await this.stateService.getDecryptedFolders();
-    if (decryptedFolders != null) {
-      return decryptedFolders;
-    }
+      if (activeAccount == null) {
+        return;
```

Comment:
Should update observables to `[]` if there's no active account

```suggestion
        await this.clear()
        return;
```